lib/phoenix/endpoint/server.ex

```diff
@@ -30,6 +29,7 @@ defmodule Phoenix.Endpoint.Server do
   defp default(config, otp_app, port) do
     config =
       config
+      |> Keyword.put_new(:handler, Phoenix.Endpoint.CowboyHandler)
```

Comment:
Instead of defining it here, just use add a key here:

https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/endpoint/adapter.ex#L108

This particular clause is only invoked for things defined dynamically.